### ISSUE TITLE
Workflows related to Calendar don't work  

### ISSUE THREAD

**Issue by xiajunwen1007** [rknightuk/alfred-workflows#7]
I have some events in my Calendar, but the workflow cannot show them and I cannot add new event to Calendar. There are some screenshots.
![屏幕录制2021-08-20 下午8 38 27](https://user-images.githubusercontent.com/49160447/130234614-580bc202-e95c-487b-84c0-ae02e0ecadcd.gif)
![屏幕录制2021-08-20 下午8 37 10](https://user-images.githubusercontent.com/49160447/130234668-e5b7f38c-cb3d-4262-abdf-8f79922c29dd.gif)

The workflow's behavior work like I have no such calendar called "alfred", but actually, I do have that.
![image](https://user-images.githubusercontent.com/49160447/130234957-170886ed-f266-41e5-80c8-24119281d25f.png)

**Comment by rknightuk** [comment]
There's a couple of things you can try here to work out what's wrong

1. Debug in Alfred

If you open Alfred, select the workflow, then click the "debug" icon in the top right you'll get logs for the workflow.
Try and run the "Upcoming" option and then see if any errors come up there.

![image](https://user-images.githubusercontent.com/948436/130249725-01ce866f-0b2d-4c96-a4ca-160282c97310.png)


2. Run it manually

If you open the directory the workflow is stored in terminal you can try to run it manually with `.agendas events` and see if you see an error there or get a permissions error.

**Comment by xiajunwen1007** [comment]
The "Upcoming" option works quite well and no error occurs. I tried the "Add new event" option, and here is the output.
```
[07:17:35.915] Logging Started...
[07:17:40.865] Agenda - Reminders and Calendar Events[Script Filter] Queuing argument '(null)'
[07:17:41.142] Agenda - Reminders and Calendar Events[Script Filter] Script with argv '(null)' finished
[07:17:41.146] Agenda - Reminders and Calendar Events[Script Filter] {"items":[{"title":"Today Overview","arg":"today","uid":"today","icon":{"path":"icons/overview.png"},"variables":{"todayOnly":true}},{"title":"Upcoming","arg":"upcoming","uid":"upcoming","icon":{"path":"icons/overview.png"},"variables":{"todayOnly":false}},{"title":"Reminder Lists","arg":"reminder_lists","uid":"reminder_lists","icon":{"path":"icons/list.png"}},{"title":"Create New Reminder","arg":"reminder_create","uid":"reminder_create","icon":{"path":"icons/add.png"}},{"title":"Calendars","arg":"calendar_lists","uid":"calendar_lists","icon":{"path":"icons/calendar.png"}},{"title":"Create New Event","arg":"calendar_create","uid":"calendar_create","icon":{"path":"icons/add.png"}},{"title":"Manage Settings","arg":"settings","uid":"settings","icon":{"path":"icons/settings.png"}}]}
[07:17:42.621] Agenda - Reminders and Calendar Events[Script Filter] Processing complete
[07:17:42.628] Agenda - Reminders and Calendar Events[Script Filter] Passing output 'calendar_create' to Arg and Vars
[07:17:42.629] Agenda - Reminders and Calendar Events[Arg and Vars] Processing complete
[07:17:42.630] Agenda - Reminders and Calendar Events[Arg and Vars] Passing output '' to Conditional
[07:17:42.631] Agenda - Reminders and Calendar Events[Conditional] Processing complete
[07:17:42.632] Agenda - Reminders and Calendar Events[Conditional] Passing output '' to Call External Trigger
[07:17:42.634] Agenda - Reminders and Calendar Events[External] Processing complete
[07:17:42.636] Agenda - Reminders and Calendar Events[External] Passing output '' to Keyword
[07:17:54.034] Agenda - Reminders and Calendar Events[Keyword] Processing complete
[07:17:54.036] Agenda - Reminders and Calendar Events[Keyword] Passing output 'Lunch tomorrow 12 until 2' to Conditional
[07:17:54.038] Agenda - Reminders and Calendar Events[Conditional] Processing complete
[07:17:54.040] Agenda - Reminders and Calendar Events[Conditional] Passing output 'Lunch tomorrow 12 until 2' to Run Script
[07:17:54.369] Agenda - Reminders and Calendar Events[Run Script] Processing complete
[07:17:54.381] Agenda - Reminders and Calendar Events[Run Script] Passing output '' to Conditional
[07:17:54.384] Agenda - Reminders and Calendar Events[Conditional] Processing complete
[07:17:54.386] Agenda - Reminders and Calendar Events[Conditional] Passing output '' to Run Script
[07:17:54.865] Agenda - Reminders and Calendar Events[Run Script] Processing complete
[07:17:54.880] Agenda - Reminders and Calendar Events[Run Script] Passing output 'No reminders list matching alfred
' to Post Notification 
```
I tried to run .agend from command line, and here is the error message
```
2021-08-21 07:26:07.816 agendas[9938:254515] CoreData: XPC: Unable to load metadata: Error Domain=NSCocoaErrorDomain Code=134070 "An error occurred in the persistent store." UserInfo={Problem=request failed, insufficient permission}
CoreData: error: addPersistentStoreWithType:configuration:URL:options:error: returned error NSCocoaErrorDomain (134070)
CoreData: annotation: userInfo:
CoreData: annotation: 	Problem : request failed, insufficient permission
CoreData: annotation: storeType: NSXPCStore
CoreData: annotation: configuration: (null)
CoreData: annotation: URL: file:///Users/yyg/Library/Calendars/Calendar%20Cache
CoreData: annotation: options:
CoreData: annotation: 	NSMigratePersistentStoresAutomaticallyOption : 1
CoreData: annotation: 	agentOrDaemon : 1
CoreData: annotation: 	NSInferMappingModelAutomaticallyOption : 1
CoreData: annotation: 	serviceName : com.apple.CalendarAgent.database
CoreData: annotation: 	NSPersistentHistoryTrackingKey : {
    NSPersistentHistoryTrackingEntitiesToExclude =     (
        ChangeRequest
    );
}

```

It seems there is a permission error. I checked wether .agend have the permission to Calendar, and the answer is no.
![image](https://user-images.githubusercontent.com/49160447/130303021-ae7fae6f-149a-452a-a4d2-f7ad2816a50e.png)

My OS version is 11.5.1, and there is no apparent way to add the permission.

I tried .agend again, then there is no error message. I give all of the permissions the .agend required at the first launch. So the error vanishes I guess.

**Comment by rknightuk** [comment]
I just released v1.0.1 thanks to @yonbergman who found the issue. New version is here https://github.com/rknightuk/alfred-workflows/tree/main/workflows/agenda
